src/hdmf_zarr/backend.py

```diff
@@ -1268,6 +1273,23 @@ def __list_fill__(self, parent, name, data, options=None):  # noqa: C901
         else:
             data_shape = get_data_shape(data)
 
+        # Let's check to see if we have a structured array somewhere in the data
+        # If we do, then we are going to resort to pickling the data and
+        # printing a warning.
+        has_structured_array = False
+        if dtype == object:
+            for c in np.ndindex(data_shape):
+                o = data
+                for i in c:
+                    o = o[i]
+                if isinstance(o, np.void) and o.dtype.names is not None:
+                    has_structured_array = True
```

Comment:
this seems pretty slow, iterating through every value in the array to check type. in structured arrays, wouldn't you just have to check the dtype of the whole array? in any case shouldn't we `break` the first time we get a `True`?

Reply:
also couldn't this switch happen here: https://github.com/hdmf-dev/hdmf-zarr/blob/29adcca1af14bb03255fdf52723daedf6c67ce44/src/hdmf_zarr/backend.py#L1254 ?